### ISSUE TITLE
NZBGet to Medusa - Failed/Bad Downloads Not Retrying

### ISSUE THREAD

**Issue by arrrghhh** [clinton-hall/nzbToMedia#1750]
**Describe the bug**
I thought initially the issue was related to NZBGet using a very outdated version of the NZBtoMedia script.  After updating, I still have issues with failed/bad download processing.

**Technical Specs**
1. Ubuntu 18.04, NZBGet and Medusa are in docker containers (on the same machine)
2. Python version '2.7'
3. Download Client: NZBget
4. Intended Media Management: Medusa

**Expected behavior**
When a download fails in NZBGet (usually due to health dropping below the threshold) there is a 'HealthCheck' process within NZBGet which deletes the download and should send this info back to Medusa to grab another.  However, this never happens - failed downloads never retry and the episode in Medusa just gets stuck on 'snatched'.

Successful downloads post-process as expected, it's only failed/bad downloads which do not post-process successfully.

**Log**
```
2020-05-24 16:48:59 WARNING ::MAIN: movie category is set for CouchPotato and Watcher3. Please check your config in NZBGet
2020-05-24 16:49:00 INFO    ::MAIN: Python v2.7 reached end of life 144 days ago.
2020-05-24 16:49:00 WARNING ::MAIN: Please upgrade to a more recent Python version.
2020-05-24 16:49:00 INFO    ::MAIN: Checking database structure...
2020-05-24 16:49:00 DEBUG   ::MAIN: Checking Initial Schema database upgrade
2020-05-24 16:49:00 DEBUG   ::MAIN: InitialSchema upgrade not required
2020-05-24 16:49:00 DEBUG   ::MAIN: Checking if we can use git commands: git version
2020-05-24 16:49:00 DEBUG   ::MAIN: Executing git version with your shell in /config/scripts
2020-05-24 16:49:00 DEBUG   ::MAIN: Not using: git
2020-05-24 16:49:00 DEBUG   ::MAIN: Unable to find your git executable - Set git_path in your autoProcessMedia.cfg OR delete your .git folder and run from source to enable updates.
2020-05-24 16:49:00 DEBUG   ::MAIN: No git specified, can't use git commands
2020-05-24 16:49:00 INFO    ::MAIN: Checking if git needs an update
2020-05-24 16:49:00 DEBUG   ::MAIN: No git specified, can't use git commands
2020-05-24 16:49:00 ERROR   ::MAIN: Unable to determine installed version via git, please check your logs!
2020-05-24 16:49:00 INFO    ::MAIN: No update needed
2020-05-24 16:49:00 INFO    ::MAIN: nzbToMedia Version:12.1.05 Branch:master (Linux 4.15.0-60-generic)
2020-05-24 16:49:00 WARNING ::MAIN: Failed to locate par2. Repair and rename using par files will not be possible!
2020-05-24 16:49:00 WARNING ::MAIN: Failed to locate ffmpeg. Transcoding disabled!
2020-05-24 16:49:00 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2020-05-24 16:49:00 WARNING ::MAIN: Failed to locate ffprobe. Video corruption detection disabled!
2020-05-24 16:49:00 WARNING ::MAIN: Install ffmpeg with x264 support to enable this feature  ...
2020-05-24 16:49:00 INFO    ::MAIN: #########################################################
2020-05-24 16:49:00 INFO    ::MAIN: ## ..::[nzbToMedia.py]::.. ##
2020-05-24 16:49:00 INFO    ::MAIN: #########################################################
2020-05-24 16:49:00 DEBUG   ::MAIN: Options passed into nzbToMedia: ['/config/scripts/nzbToMedia.py']
2020-05-24 16:49:00 INFO    ::MAIN: Script triggered from NZBGet Version 21.0.
2020-05-24 16:49:00 INFO    ::MAIN: Download failed with status FAILURE/HEALTH.
2020-05-24 16:49:00 DEBUG   ::MAIN: Adding NZB download info for directory /incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9432 to database
2020-05-24 16:49:00 INFO    ::MAIN: Auto-detected SECTION:SickBeard
2020-05-24 16:49:00 DEBUG   ::MAIN: Checking for archives to extract in directory: /incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9432
2020-05-24 16:49:00 INFO    ::MAIN: Calling SickBeard:tv to post-process:Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee
2020-05-24 16:49:00 DEBUG   ::SERVER: Attempting to connect to server at http://host:8081
2020-05-24 16:49:00 DEBUG   ::SERVER: Server responded at http://host:8081
2020-05-24 16:49:00 INFO    ::MAIN: Attempting to auto-detect tv fork
2020-05-24 16:49:00 INFO    ::MAIN: SickBeard:tv fork auto-detection found custom params {u'force': None, u'is_priority': None, u'failed': None, u'process_method': None, u'proc_dir': None, u'type': None, u'delete_on': None}
2020-05-24 16:49:00 INFO    ::MAIN: SickBeard:tv fork set to custom
2020-05-24 16:49:00 WARNING ::SICKBEARD: No media files found in directory /incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9432. Processing this as a failed download
2020-05-24 16:49:00 POSTPROCESS::SICKBEARD: FAILED: The download failed. Sending 'failed' process request to custom branch
2020-05-24 16:49:00 DEBUG   ::SICKBEARD: Opening URL: http://host:8081/home/postprocess/processEpisode with params: {u'nzbName': 'Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee', u'quiet': 1, u'failed': 1, u'proc_dir': '/incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9432', u'type': u'manual'}
2020-05-24 16:49:00 POSTPROCESS::SICKBEARD: Post-processing attempted but directory doesn't exist: /incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9432
2020-05-24 16:49:00 ERROR   ::MAIN: A problem was reported in the /config/scripts/nzbToMedia.py script.
```

**Comment by arrrghhh** [comment]
Hm, I switched to using an API key and I no longer get that error at the end but Medusa still doesn't try a new download...

**Comment by clinton-hall** [comment]
What do Medusa logs show at `2020-05-24 18:06:52`?
The script tells Medusa that the episode failed, So if Medusa doesn't like the api command etc, it should log some error to explain why.

**Comment by clinton-hall** [comment]
same as #1528

**Comment by arrrghhh** [comment]
Hi clinton-hall, thanks so much for the reply.  This fell off my radar, but I was thankfully able to grab logs back to this date (of course Medusa seems to log in local time and nzbToMedia in UTC, but that's fine):

```
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: 192.168.0.1 - gave correct API KEY. ACCESS GRANTED
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: all args: ()
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: all kwargs: "{'cmd': 'sg.postprocess', 'help': '1'}"
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: sg.postprocess: {'help': '1'}
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: 192.168.0.1 - gave correct API KEY. ACCESS GRANTED
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: all args: ()
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: all kwargs: "{'cmd': 'help', 'subject': 'postprocess'}"
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: help: {'subject': 'postprocess'}
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: 192.168.0.1 - gave correct API KEY. ACCESS GRANTED
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: all args: ()
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: all kwargs: "{'nzbName': 'Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee', 'failed': '1', 'cmd': 'postprocess', 'path': '/incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9435', 'type': 'manual', 'return_data': '0'}"
2020-05-24 12:06:52 DEBUG    TORNADO :: [] API :: postprocess: {'nzbName': 'Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee', 'failed': '1', 'path': '/incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9435', 'type': 'manual', 'return_data': '0'}
2020-05-24 12:06:52 WARNING  TORNADO :: [] Post-processing attempted but directory doesn't exist: /incomplete/Real.Time.With.Bill.Maher.S18E14.May.8.2020.720p.AMZN.WEB-DL.DD.2.0.H.264-monkee.#9435
```
So that warning is correct, that directory doesn't exist - because the download failed and NZBGet removed it.

Perhaps there's something basic I'm missing in Medusa?  I've tried to get help from that side as well, since I thought Medusa was the problem child in all of this... if you feel the same I will try to push my issue over on that side.

**Comment by clinton-hall** [comment]
> So that warning is correct, that directory doesn't exist - because the download failed and NZBGet removed it.

Is there an option for NZBGet to NOT delete failed? 
If the directory still exists, then Medusa should process (as failed), re-snatch, and delete the incomplete files...

I believe nzbToMedia also does a cleanup check at the end for failed downloads (if not I can certainly add something...) So this should keep it clean...

**Comment by arrrghhh** [comment]
So as far as I know it is enabled.  These are the options for 'HealthCheck' in NZBGet:

> Delete - delete nzb-file from queue, also delete already downloaded files;
> Park - move nzb-file to history, keep already downloaded files. Commands "Download remaining files" and "Retry failed articles" are available for this nzb;
> Pause - pause nzb-file;
> None - do nothing (continue download).

I originally had it set to 'Delete' but I have recently set it to 'Park'.  I'll try it again just to make sure, but I'm thinking perhaps Medusa can't "see" that incomplete folder... I thought I fixed that issue a while ago but perhaps it came back (sometimes my docker container management is not the best!)

**Comment by arrrghhh** [comment]
Gah, that was it.  I updated a script locally for Medusa, but I had another docker container update script on the machine itself... of course I forgot to update that script, so the mountpoint for the "/incomplete" folder was no longer there.

So long story short, I misconfigured Medusa as it was not able to get to the /incomplete folder.  Even setting the HealthCheck to 'Delete' in NZBGet, now retry works!!

Sorry for the trouble, appreciate the help immensely.  Closing this...
